cpp/dolfinx/geometry/utils.cpp

```diff
@@ -29,16 +29,19 @@ constexpr bool is_leaf(const std::array<int, 2>& bbox)
 /// A point `x` is inside a bounding box `b` if each component of its
 /// coordinates lies within the range `[b(0,i), b(1,i)]` that defines the bounds
 /// of the bounding box, b(0,i) <= x[i] <= b(1,i) for i = 0, 1, 2
-constexpr bool point_in_bbox(const std::array<std::array<double, 3>, 2>& b,
-                             const std::array<double, 3>& x)
+constexpr bool point_in_bbox(std::span<const double, 6> b,
+                             std::span<const double, 3> x)
 {
+  assert(b.size() == 6);
   constexpr double rtol = 1e-14;
   bool in = true;
+  auto b0 = b.subspan(0, 3);
```

Comment:
Does this syntax make the span size a compile-time constant?

There are examples at https://en.cppreference.com/w/cpp/container/span/dynamic_extent on how to check.

Reply:
We should use 
```c++
b.subspan<0,3>
```
As I tested it with:
```c++
td::vector<double> vec(10);
  std::span<double> sp(vec);
  print("span", sp.extent);
  constexpr std::size_t start = 0;
  constexpr std::size_t end = 3;

  auto ssp = sp.subspan(start, end);
  print("subspan", ssp.extent);
  std::span<double, 3> sssp = sp.subspan<0, 3>();
  print("constexpr subspan", sssp.extent);
```
yielding
```bash
span, dynamic extent
subspan, dynamic extent
constexpr subspan, static extent = 3
```
See https://en.cppreference.com/w/cpp/container/span/subspan for signature details. I'll fix this later today or tomorrow.